### ISSUE TITLE
Why is this abandoned?

### ISSUE THREAD

**Issue by appsforartists** [facebookarchive/react-page#58]
@jordwalke + @zpao 

There's a fat banner on the top of your readme that says:

> This project is not actively maintained. Proceed at your own risk!

While I appreciate the warning, more detail would be helpful.  Did you reach a dead-end while working on `react-page`?  Is there a similar project that better solves this problem?  Is it simply a matter of not having the resources to continue dev?  Would you recommend others pick up where you left off?

Answers to some of these questions would be really helpful in understanding the best practices for using React on the server.  (The SEO benefit from React's client/server agnosticism is a huge selling point for my team.)

**Comment by appsforartists** [comment]
CCing @spicyj, who replied to a [similar comment on React](https://github.com/facebook/react/issues/1519#issuecomment-51415311)

**Comment by jordwalke** [comment]
> > Did you reach a dead-end while working on react-page?

No, no technical dead ends. It was just a matter of not having the time for a while. I still use it personally for rapid reloading of pages and we still use it to develop documentation sites for projects such as Jest. I built it because there was no good one-click solution for a development flow that would allow rapid reloading (without watching) in under 50 ms, with server rendering capabilities and source maps support out of the box. It was intended to serve as a //minimum// developer experience bar. IMHO, everything else that people build for a react dev environment should be better than this, though it's not always true. Since originally building react-page, several other solutions have been built that implement many of these features. I'm not sure if they are totally comprehensive. Also, `react-page-middleware/node-haste` support the `@providesModule` dockblock directive which the React core itself uses so that means I could develop the React core with react-page-middleware - no other bundler/resource analyzer understands that convention.

> > Is there a similar project that better solves this problem?

I'm not sure, but I believe react-app might do most of this. I hear good things about webpack too. If anyone knows of another rapid reload environment, please let me know.

> > Is it simply a matter of not having the resources to continue dev? Would you recommend others pick up where you left off?

It really is just a matter of development time. I would have continued it, and I might pick it back up one day. But I hope by then, others will have built a one-click solution that does all of this and more.

If I were to continue this, I would split `react-page-middleware` into two pieces:
- One is a simple source-mapping bundler: The one in `react-page-middleware` is really good because it uses `node-haste` which exposes a ton of power and ability to do fine grained caching. It transforms all of the ES6 features on the fly without running a watcher. Watchers are personal pet peeves of mine because I remap CMD+r to save the file and reload the page directly from Vim, and watchers typically can't get to the file in time causing out of date page loads. Performance is important to me.
- Server rendering route handler: I would separate this from the source-map/transforming/bundling features completely (even a different npm package).
- I would also use webpack if possible.
- I would allow specifying of incremental module loading, and splitting up bundles into segments so you don't need to recompute the entire bundle. I'm sure webpack supports this.

**Comment by jordwalke** [comment]
Another thing to note is that `react-page` was just supposed to be a //standard// of bare minimum features that doesn't specify how they're implemented. If something else has all of these features, then we should just replace all of the `react-page` implementation with that project or just redirect people to it. IMHO, it doesn't matter how we get all of these features, it just matters that we have a one click solution that provides them. Again, here's that list:
- Extremely fast reloading. (Under 50ms for reasonably large projects). (Likely requires caching)
- Source Maps (options for either inline or linked maps).
- Ability to specify our standard ES6/JSX transforms.
- A single `npm install` command away. ("one click").
- Server rendering (a plus).
- Understands @providesModule (just to develop the React core itself)
- Routing that allow rendering of entire page in React.
- Example routing that maps a single page to a single component that is expected to render the `<html>` tags etc.

**Comment by appsforartists** [comment]
Thanks for the thorough reply.  :smiley:
